include/envoy/http/filter.h

```diff
@@ -189,6 +189,12 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    */
   virtual const Buffer::Instance* decodingBuffer() PURE;
 
+  /**
+   * Allows modifying the decoding buffer. May only be called before any data has been continued
+   * past the calling filter.
+   */
+  virtual void modifyDecodingBuffer(std::function<void(Buffer::Instance&)> callback) PURE;
```

Comment:
Used a callback style here to make it clear to the caller that they're not supposed to retain a pointer to the buffer